numpy/lib/tests/test_loadtxt.py

```diff
@@ -1043,3 +1043,13 @@ def test_field_growing_cases():
     for i in range(1, 1024):
         res = np.loadtxt(["," * i], delimiter=",", dtype=bytes)
         assert len(res) == i+1
+
+@pytest.mark.parametrize("nmax", (10000, 50000, 80000, 100000, 120000)) # less, equal, greater, twice and more than twice than _loadtxt_chunksize
```

Comment:
```suggestion
@pytest.mark.parametrize("nmax", (10000, 50000, 80000, 100000, 120000)) # less, equal, greater, twice and more than twice than _loadtxt_chunksize
```
The line is long and, maybe move the comment (even in the function body).

Reply:
> while on a file object it misses end lines, causing also the right units in test_parametric_unit_discovery, because it should infer them from the last line...

I think there's actually a bug in the test itself - the `nrows` parameter from the `test_parametric_unit_discovery` doesn't appear to actually be used. If you add `max_rows=nrows` to the two calls to loadtxt, you see failures when `nrows` is <= 50000 (i.e. the _loadtxt_chunksize). Perhaps this is what you were saying above (where the "last line" is dictated by the value of `nrows`). However, I'd argue that's actually *correct* behavior - shouldn't the dtype only be determined by the values in the chunk requested to be read by the users?